### ISSUE TITLE
HasAlternateKey

### ISSUE THREAD

**Issue by TimSirmovics** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#545]
When a unique key is defined on a table:
![image](https://user-images.githubusercontent.com/10243286/68437056-e7835700-0213-11ea-8ab5-8fd8d343c56f.png)

The following is generated in the `Configure` method:
`builder.HasIndex(x => x.AltKey).HasName("IX_AltKey").IsUnique();`

Could we also add a `HasAlternateKey()` call just after where we have the `HasKey()` call?
```
builder.ToTable("TableName", "dbo");
builder.HasKey(x => x.PrimaryKey);
builder.HasAlternateKey(@"AltKey");
```

This is a request for the V3 generator.

**Comment by TimSirmovics** [comment]
I just realised this could also be added as an expression like the primary key: `builder.HasAlternateKey(x => x.AltKey);`

**Comment by sjh37** [comment]
Thanks @TimSirmovics 
It's on my todo list.

**Comment by sjh37** [comment]
Hi @TimSirmovics Can you take a look at the latest [EF.Reverse.POCO.v3.ttinclude](https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/master/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude) file and let me know what you think.

I've tested it against all the test databases I use, and would value your opinion.

**Comment by sjh37** [comment]
Found a fault. I can't include columns in HasAlternateKey if they are allowed to be nullable.

**Comment by sjh37** [comment]
Hi @TimSirmovics Can you take a look at the latest [EF.Reverse.POCO.v3.ttinclude](https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/master/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude) file and let me know what you think.

I've tested it against all the test databases I use, and would value your opinion.

**Comment by TimSirmovics** [comment]
Hi @sjh37,

A basic look at the generated output appears ok.

My use case for this feature was going to tie in with a feature I had requested in another EF Library [reconciler](https://github.com/jtheisen/reconciler).
Unfortunately since I opened this issue to update the generator, the developer of that library has noted he won't be developing it any further, so at this stage I don't have any code that requires this feature and can't really test it properly.
I have forked the reconciler project and may in future develop it myself, and if that is the case I will most likely need this so happy to leave it in, I just can't provide much real world testing sorry.

**Comment by brianrutter** [comment]
For me, it looks like in v3.1.0 it is putting "HasAlternateKey" for all indexes, not just unique indexes. I think it is supposed to be for only unique indexes.

EfCore3.

 Let me know if you need more info.

**Comment by sjh37** [comment]
@brianrutter @TimSirmovics I have had to remove the use of `HasAlternateKey` due to the fact it marks the properties as read-only. Adding a unique constraint is done via `HasIndex`, and using `HasAlternateKey` has the same effect, but also marks the properties as read-only.

This will be in the next release. If you want the code now, just grab the latest [EF.Reverse.POCO.v3.ttinclude](https://raw.githubusercontent.com/sjh37/EntityFramework-Reverse-POCO-Code-First-Generator/master/EntityFramework.Reverse.POCO.Generator/EF.Reverse.POCO.v3.ttinclude) file.

**Comment by sjh37** [comment]
Released in v3.1.1
